.github/workflows/ci.yml

```diff
@@ -147,6 +147,10 @@ jobs:
           # brew update sometimes fails with "Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask failed!"
           brew update || brew update-reset
           brew install autoconf automake libtool
+      - name: Install valgrind
+        if: ${{ matrix.arch == 'amd64' }}
+        run: |
+          brew install valgrind
```

Comment:
valgrind works on OS X?

Reply:
No, I deleted the wrong hunk!
But initially I thought `valgrind` worked also on macOS, apparently not. :/

Reply:
@emanuele6 - On Mac, `brew install valgrind` claims it only works on Linux, but there is a valgrind variant that works on Mac OS:
```
brew tap LouisBrunner/valgrind; brew install --HEAD LouisBrunner/valgrind/valgrind
```
When I ran test/shtest, it reveals the following:
```
+ valgrind --error-exitcode=1 --leak-check=full --suppressions=/Users/user/github/jqlang/tests/onig.supp --suppressions=/Users/user/github/jqlang/tests/local.supp -q /Users/user/github/jqlang/tests/../jq -Rse '. == "a\u0000b\nc\u0000d\ne"' /var/folders/15/wd4cntyn53x87v4x86wxysyc0000gn/T/jqXXXXXX.koOzeC6M/input
--57220-- run: /usr/bin/dsymutil "/Users/user/github/jqlang/tests/../jq"
==57220== Conditional jump or move depends on uninitialised value(s)
==57220==    at 0x7FF80BB3EA3F: ???
==57220==    by 0x7FF80BA1A8ED: fgets (in /usr/lib/system/libsystem_c.dylib)
==57220==    by 0x10003142E: jq_util_input_read_more (util.c:305)
==57220==    by 0x100030F67: jq_util_input_next_input (util.c:411)
==57220==    by 0x1000029D7: main (main.c:690)
==57220== 
true
==57220== 24 bytes in 1 blocks are possibly lost in loss record 4 of 30
....
==57220== 1,888 bytes in 59 blocks are definitely lost in loss record 30 of 30
```

Can you tell whether this is a jq issue, or a valgrind-on-a-Mac issue?

Reply:
@pkoppstein `"Conditional jump or move depends on uninitialised value(s)"` for a libc function like `fgets()`, likely just means that you are running a static build of `jq` with `valgrind`, and `valgrind` doesn't work very well with static builds.